modin/experimental/xgboost/xgboost.py

```diff
@@ -0,0 +1,482 @@
+        `evals_result`, which is returned as part of function return value
+        instead of argument.
+
+    Returns
+    -------
+    dict
+        A dictionary containing trained booster and evaluation history.
+        `history` field is the same as `eval_result` from `xgboost.train`.
+
+        .. code-block:: python
+
+            {'booster': xgboost.Booster,
+             'history': {'train': {'logloss': ['0.48253', '0.35953']},
+                         'eval': {'logloss': ['0.480385', '0.357756']}}}
+    """
+    LOGGER.info("Training started")
+
+    s = time.time()
+
+    X, y = dtrain
+    assert len(X) == len(y)
+    X_row_parts = unwrap_row_partitions(X, bind_ip=not evenly_data_distribution)
+    y_row_parts = unwrap_row_partitions(y, bind_ip=not evenly_data_distribution)
+    assert len(X_row_parts) == len(y_row_parts), "Unaligned train data"
+
+    # Create remote actors
+    actors = create_actors(nthread=nthread)
+
+    add_as_eval_method = None
+    if evals:
+        for (eval_data, method) in evals[:]:
+            if eval_data is dtrain:
+                add_as_eval_method = method
+                evals.remove((eval_data, method))
+
+        for ((eval_X, eval_y), eval_method) in evals:
+            # Split data across workers
+            _split_data_across_actors(
+                actors,
+                lambda actor, *Xy: actor.add_eval_data.remote(
+                    *Xy, eval_method=eval_method
+                ),
+                unwrap_row_partitions(eval_X, bind_ip=not evenly_data_distribution),
+                unwrap_row_partitions(eval_y, bind_ip=not evenly_data_distribution),
+                evenly_data_distribution=evenly_data_distribution,
+            )
+
+    # Split data across workers
+    _split_data_across_actors(
+        actors,
+        lambda actor, *Xy: actor.set_train_data.remote(
+            *Xy, add_as_eval_method=add_as_eval_method
+        ),
+        X_row_parts,
+        y_row_parts,
+        evenly_data_distribution=evenly_data_distribution,
+    )
+    LOGGER.info(f"Data preparation time: {time.time() - s} s")
+
+    s = time.time()
+    # Start Rabit tracker
+    env = _start_rabit_tracker(len(actors))
+    rabit_args = [("%s=%s" % item).encode() for item in env.items()]
+
+    # Train
+    fut = [actor.train.remote(rabit_args, params, *args, **kwargs) for actor in actors]
+
+    # All results should be the same because of Rabit tracking. So we just
+    # return the first one.
+    result = ray.get(fut[0])
+    LOGGER.info(f"Training time: {time.time() - s} s")
+    LOGGER.info("Training finished")
+    return result
+
+
+def predict(
+    model,
+    data: ModinDMatrix,
+    nthread: Optional[int] = cpu_count(),
+    evenly_data_distribution: Optional[bool] = True,
+    **kwargs,
+):
+    """
+    Run prediction with a trained booster.
+
+    Parameters
+    ----------
+    model: A Booster or a dictionary returned by `modin.experimental.xgboost.train`.
```

Comment:
`model:` -> `model :` and below in such way

Reply:
fixed